trim_lcnaf.rb

```diff
@@ -0,0 +1,28 @@
+File.open("trimmed_lcnaf.yml", "w") do |trimmed_file|
+  previous_id = nil
+
+  File.open("lcnaf.skos.ndjson").each_line do |line|
+    line_hash = eval(line)
+    graph = line_hash[:"@graph"]
+    term = ""
+    id = ""
+
+    graph.each do |graph_entry|
+      if graph_entry[:"@type"] == "skos:Concept"
+        # set the skos:prefLabel
+        term = graph_entry[:"skos:prefLabel"]
+        id = graph_entry[:@id]
+      elsif graph_entry[:"@type"] == "cs:ChangeSet"
+        id = graph_entry[:"cs:subjectOfChange"][:@id]
+      end
+
+      # don't allow blanks, skip repeats
+      if previous_id != id && id != ""
+        trimmed_file << "\"#{term}\":\"#{id}\"\n"
+      end
+
+      # to avoid repeated data
+      previous_id = id
```

Comment:
Let's take this representative entry (a prettified line from the ndjson file):

```json
{
  "@context": {
    "cs": "http://purl.org/vocab/changeset/schema#",
    "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
    "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
    "rdfs1": "http://www.w3.org/1999/02/22-rdf-schema#",
    "skos": "http://www.w3.org/2004/02/skos/core#",
    "skosxl": "http://www.w3.org/2008/05/skos-xl#",
    "xsd": "http://www.w3.org/2001/XMLSchema#",
    "about": "http://id.loc.gov/authorities/names/n83166666"
  },
  "@graph": [
    {
      "@id": "http://id.loc.gov/authorities/names/n83166666",
      "@type": "skos:Concept",
      "skos:altLabel": "Huang, Yao-huang",
      "skos:changeNote": [
        {
          "@id": "_:N7fa7ac18e5ca4396963dacafeaba3535"
        },
        {
          "@id": "_:N937a169f5767452484056e1d80aec4b2"
        }
      ],
      "skos:exactMatch": {
        "@id": "http://viaf.org/viaf/sourceID/LC%7Cn++83166666#skos:Concept"
      },
      "skos:inScheme": {
        "@id": "http://id.loc.gov/authorities/names"
      },
      "skos:prefLabel": "Huang, Yaohuang",
      "skosxl:altLabel": {
        "@id": "_:Nc4f7d95d194b454298110c124741db47"
      }
    },
    {
      "@id": "_:N7fa7ac18e5ca4396963dacafeaba3535",
      "@type": "cs:ChangeSet",
      "cs:changeReason": "revised",
      "cs:createdDate": {
        "@type": "xsd:dateTime",
        "@value": "2008-12-10T07:16:30"
      },
      "cs:creatorName": {
        "@id": "http://id.loc.gov/vocabulary/organizations/ocolc"
      },
      "cs:subjectOfChange": {
        "@id": "http://id.loc.gov/authorities/names/n83166666"
      }
    },
    {
      "@id": "_:Nc4f7d95d194b454298110c124741db47",
      "@type": "skosxl:Label",
      "skosxl:literalForm": "Huang, Yao-huang"
    },
    {
      "@id": "_:N937a169f5767452484056e1d80aec4b2",
      "@type": "cs:ChangeSet",
      "cs:changeReason": "new",
      "cs:createdDate": {
        "@type": "xsd:dateTime",
        "@value": "1983-12-13T00:00:00"
      },
      "cs:creatorName": {
        "@id": "http://id.loc.gov/vocabulary/organizations/dlc"
      },
      "cs:subjectOfChange": {
        "@id": "http://id.loc.gov/authorities/names/n83166666"
      }
    }
  ]
}
``` 

- There is only 1 term per ndjson line that is identified. The `id` is always given by `line_hash["@context"]["about"]`. 
- The `"cs:ChangeSet"` type indicates there was a change, but the change is already reflected in the entry with "skos:Concept". The entries before it had a different label. Let's not worry about historical changes and just take the most recent one for now. It's a reasonable trade-off to get things done (it's actually a feature in some respect: we made all the terms in use current LCNAF terms!). 
- In my experience, I've encountered cases where `@type` was `"skosxl:Label"` (there was no `skos:Concept`) and the label (`term`) was `"skosxl:literalForm"`
- The other elements of the `"@graph"`  are skos-specific graph annotations. This should allow us to remove the "don't allow blanks/repeat part" and return immediately. This might improve the scanning speed too.
- if the prettified example was helpful, maybe a test with it is useful :smile_cat:

Reply:
Prettified version is very helpful. I think sometimes when I'm heads down I don't think to transform the data to something that makes more sense.